lib/ads/interstitial_ad_manager.js

```diff
@@ -214,41 +214,52 @@ shaka.ads.InterstitialAdManager = class {
       return;
     }
     this.updatePlayerConfig_();
-    let cuepointsChanged = false;
     const interstitialsAd = await this.getInterstitialsInfo_(interstitial);
     if (interstitialsAd.length) {
       for (const interstitialAd of interstitialsAd) {
-        const interstitialId = JSON.stringify(interstitialAd);
-        if (this.interstitialIds_.has(interstitialId)) {
-          continue;
-        }
-        cuepointsChanged = true;
-        this.interstitialIds_.add(interstitialId);
-        this.interstitials_.add(interstitialAd);
-        let shouldPreload = false;
-        if (interstitial.pre && this.lastTime_ == null) {
-          shouldPreload = true;
-        } else if (interstitialAd.startTime == 0 && !interstitialAd.canJump) {
-          shouldPreload = true;
-        } else if (this.lastTime_ != null) {
-          const difference = interstitial.startTime - this.lastTime_;
-          if (difference > 0 && difference <= 10) {
-            shouldPreload = true;
-          }
-        }
-        if (shouldPreload) {
-          if (!this.preloadManagerInterstitials_.has(interstitialAd)) {
-            this.preloadManagerInterstitials_.set(
-                interstitialAd, this.player_.preload(interstitialAd.uri));
-          }
-        }
+        this.addInterstitial(interstitialAd);
       }
     } else {
       shaka.log.alwaysWarn('Unsupported interstitial', interstitial);
     }
-    if (cuepointsChanged) {
-      this.cuepointsChanged_();
+  }
+
+
+  /**
+   * @param {shaka.extern.AdInterstitial} interstitial
+   */
+  addInterstitial(interstitial) {
+    if (this.basePlayer_.getLoadMode() == shaka.Player.LoadMode.SRC_EQUALS &&
+        this.usingBaseVideo_) {
+      shaka.log.alwaysWarn(
+          'Unsupported interstitial when using single media element',
+          interstitial);
+      return;
+    }
+    const interstitialId = JSON.stringify(interstitial);
+    if (this.interstitialIds_.has(interstitialId)) {
+      return;
+    }
+    this.interstitialIds_.add(interstitialId);
+    this.interstitials_.add(interstitial);
+    let shouldPreload = false;
+    if (interstitial.pre && this.lastTime_ == null) {
+      shouldPreload = true;
+    } else if (interstitial.startTime == 0 && !interstitial.canJump) {
+      shouldPreload = true;
+    } else if (this.lastTime_ != null) {
+      const difference = interstitial.startTime - this.lastTime_;
+      if (difference > 0 && difference <= 10) {
+        shouldPreload = true;
+      }
+    }
+    if (shouldPreload) {
+      if (!this.preloadManagerInterstitials_.has(interstitial)) {
+        this.preloadManagerInterstitials_.set(
+            interstitial, this.player_.preload(interstitial.uri));
+      }
     }
+    this.cuepointsChanged_();
```

Comment:
Doesn't this mean that, if a given HLS interstitial is translated into multiple ad interstitials, we will now fire one `cuepoints` event per interstitial? Instead of a single event for the entire HLS interstitial.

It might be better to instead make this method `addInterstitials`, and have it take in an array of interstitials that are all added at once. In order to change the outward behavior as little as possible, while still being useful for the `addCustomInterstitial` method.

Reply:
ok, I'll do it

Reply:
Done!